### PR TITLE
fix: Serato Metadata checbox behaviour

### LINKED ISSUE #11226: Write Serato Metadata checkbox is ignored when writing track metadata is disabled
### Bug Description

I think it would be reasonable to disable and gray out the checkbox in this case. 

### Version

2.3.3

### OS

_No response_

### PR DESCRIPTION
**Pull request by tech-bash**
Fix: #11226

Its working perfectly as wanted.

gsoc2

### CODE REVIEW

File: src/preferences/dialog/dlgpreflibrary.cpp [hunk 1]
```diff
@@ -197,7 +197,7 @@ void DlgPrefLibrary::slotResetToDefaults() {
             kHistoryTrackDuplicateDistanceDefault);
     spinbox_history_min_tracks_to_keep->setValue(1);
     checkBox_SyncTrackMetadata->setChecked(false);
-    checkBox_SeratoMetadataExport->setChecked(false);
+    checkBox_SeratoMetadataExport->setChecked(true);
```
Comment (uklotzde) [inline src/preferences/dialog/dlgpreflibrary.cpp:200]: Serato metadata export should not be enabled by default. Please revert this unrelated change.

File: src/preferences/dialog/dlgpreflibrary.cpp [hunk 2]
```diff
@@ -231,7 +231,8 @@ void DlgPrefLibrary::slotUpdate() {
     checkBox_SyncTrackMetadata->setChecked(
             m_pConfig->getValue(kSyncTrackMetadataConfigKey, false));
     checkBox_SeratoMetadataExport->setChecked(
-            m_pConfig->getValue(kSyncSeratoMetadataConfigKey, false));
+            m_pConfig->getValue(kSyncSeratoMetadataConfigKey, true));
```
Comment (uklotzde) [inline src/preferences/dialog/dlgpreflibrary.cpp:234]: Same here, please revert.

**Review by uklotzde — CHANGES_REQUESTED** [review]

File: src/preferences/dialog/dlgpreflibrary.cpp (hunk 2 above)
Comment (tech-bash) [reply src/preferences/dialog/dlgpreflibrary.cpp:234]: Ohh these were from my earlier commit(wrong one) forgot to revert it!

File: src/preferences/dialog/dlgpreflibrary.cpp [hunk 3]
```diff
@@ -553,5 +554,9 @@ void DlgPrefLibrary::updateSearchLineEditHistoryOptions() {
 void DlgPrefLibrary::slotSyncTrackMetadataToggled() {
     if (isVisible() && checkBox_SyncTrackMetadata->isChecked()) {
         mixxx::DlgTrackMetadataExport::showMessageBoxOncePerSession();
+        checkBox_SeratoMetadataExport->setEnabled(true);
+    } else {
+        checkBox_SeratoMetadataExport->setEnabled(false);
+        checkBox_SeratoMetadataExport->setChecked(false);
```
[the remark this answers is not in the document]
Comment (tech-bash) [reply src/preferences/dialog/dlgpreflibrary.cpp:560]: Ok.

File: src/preferences/dialog/dlgpreflibrary.cpp [hunk 4]
```diff
@@ -232,6 +232,7 @@ void DlgPrefLibrary::slotUpdate() {
             m_pConfig->getValue(kSyncTrackMetadataConfigKey, false));
     checkBox_SeratoMetadataExport->setChecked(
             m_pConfig->getValue(kSyncSeratoMetadataConfigKey, false));
+    checkBox_SeratoMetadataExport->setEnabled(false);
```
Comment (uklotzde) [inline src/preferences/dialog/dlgpreflibrary.cpp:235]: This line is wrong and has to be replaced with

```
checkBox_SeratoMetadataExport->setEnabled(checkBox_SyncTrackMetadata->isChecked());
```

**Review by uklotzde — CHANGES_REQUESTED** [review]
This could have been a single line fix. Almost. Unfortunately, the legacy code requires to duplicate the fix line.

The actual challenge is to rewrite the UI logic in a way to avoid the error prone duplication.

File: src/preferences/dialog/dlgpreflibrary.cpp [hunk 5]
```diff
@@ -553,5 +554,8 @@ void DlgPrefLibrary::updateSearchLineEditHistoryOptions() {
 void DlgPrefLibrary::slotSyncTrackMetadataToggled() {
     if (isVisible() && checkBox_SyncTrackMetadata->isChecked()) {
         mixxx::DlgTrackMetadataExport::showMessageBoxOncePerSession();
+        checkBox_SeratoMetadataExport->setEnabled(true);
```
[the remark this answers is not in the document]
Comment (uklotzde) [reply src/preferences/dialog/dlgpreflibrary.cpp:557]: You should first and foremost aim for correctness, not efficiency.

[the remark this answers is not in the document]
Comment (tech-bash) [reply src/preferences/dialog/dlgpreflibrary.cpp]: Yes, will keep that in mind while coding for mixxx.

**Comment by uklotzde** [comment]
Whoever is going to merge this:

- Switch target to 2.3 or 2.4
- Squash commits to get rid of the unrelated changes in the history

**Comment by tech-bash** [comment]
> Whoever is going to merge this:
> 
> * Switch target to 2.3 or 2.4
> * Squash commits to get rid of the unrelated changes in the history
Should I squash and change the branch myself?

**Comment by uklotzde** [comment]
@tech-bash https://github.com/RomuloOliveira/commit-messages-guide/blob/master/README.md (just one out of many resources)

**Comment by tech-bash** [comment]
I myself squashed commits.

**Comment by tech-bash** [comment]
Now, is it ready for merge?

**Comment by Swiftb0y** [comment]
Unfortunately not, you just merged but you need to rebase onto 2.4 instead.

File: src/preferences/dialog/dlgpreflibrary.cpp [hunk 6]
```diff
@@ -553,5 +554,6 @@ void DlgPrefLibrary::updateSearchLineEditHistoryOptions() {
 void DlgPrefLibrary::slotSyncTrackMetadataToggled() {
     if (isVisible() && checkBox_SyncTrackMetadata->isChecked()) {
         mixxx::DlgTrackMetadataExport::showMessageBoxOncePerSession();
+        checkBox_SeratoMetadataExport->setEnabled(checkBox_SyncTrackMetadata->isChecked());
```
Comment (daschuer) [inline src/preferences/dialog/dlgpreflibrary.cpp:557]: Why is this in this if branch? I also do not understand what the "isVisible()" check is for. 
Does this guard against the initial call? 

You can debug it by placing qDebug() statements in this function. They are printed into the console if you start mixxx with "./mixxx --developer"

**Review by daschuer — CHANGES_REQUESTED** [review]

File: src/preferences/dialog/dlgpreflibrary.cpp (hunk 6 above)
Comment (uklotzde) [reply src/preferences/dialog/dlgpreflibrary.cpp:557]: Your question is valid. But unrelated and does not matter in the context of this PR.

Comment (daschuer) [reply src/preferences/dialog/dlgpreflibrary.cpp:557]: For my understanding this is wrong.

Comment (ronso0) [reply src/preferences/dialog/dlgpreflibrary.cpp:557]: Correct me if I'm wrong, but IIUC this prevents the message box to be shown when the checkbox is updated (e.g. in `slotUpdate`?) while DlgPrefLibrary is not shown, e.g. during consctruction.

Comment (uklotzde) [reply src/preferences/dialog/dlgpreflibrary.cpp:557]: Please stop discussing unrelated questions in a PR. Thank you.

Comment (daschuer) [reply src/preferences/dialog/dlgpreflibrary.cpp:557]: Why do you think this is unrelated. The code is wrong, we need to set the state unconditionally.

Comment (uklotzde) [reply src/preferences/dialog/dlgpreflibrary.cpp:557]: @tech-bash Please fix your code as originally suggested.

Comment (tech-bash) [reply src/preferences/dialog/dlgpreflibrary.cpp]: I will open a seperate PR.
